src/DependencyInjection/CompilerPass/SubscriberPass.php

```diff
@@ -22,15 +23,21 @@
  */
 class SubscriberPass implements CompilerPassInterface
 {
+    const FACTORY_SERVICE_ID = 'csa_guzzle.client_factory';
+    const SUBSCRIBER_TAG = 'csa_guzzle.subscriber';
+    const CLIENT_TAG = 'csa_guzzle.client';
+
     public function process(ContainerBuilder $container)
     {
-        $subscribers = $container->findTaggedServiceIds('csa_guzzle.subscriber');
+        $this->addSubscribersToClients($container);
+
+        $subscribers = $container->findTaggedServiceIds(self::SUBSCRIBER_TAG);
```

Comment:
Rather than fetching these twice (Here, and in addSubscribersToClients), I would reuse the call to findTaggedServiceIds(self::SUBSCRIBER_TAG), and pass it as an argument in addSubscribersToClients.